lib/madness/document.rb

```diff
@@ -67,6 +67,12 @@ def set_base_attributes_for_directory
 
     def markdown
       @markdown ||= File.read file
+      @markdown = shortlinks(@markdown) if config.shortlinks
+      @markdown
+    end
+
+    def shortlinks(raw)
+      raw.gsub(/\[\[([^\]]+)\]\]/) { "[#{Regexp.last_match(1)}](./#{Regexp.last_match(1).to_href}.md)" }
     end
```

Comment:
There are probably some ways to optimize and/or make this more readable.
Also - I am somewhat concerned it will also replace undesirable replacements, like in code blocks.

Reply:
The merge request was intended to show a proof of principle. I have zero experience programming in Rust. So while I agree on the points you make in the requests for changes, I don't believe I'm in the position to improve the code effectively to reach common standards - I'd have to properly learn Rust first.

I wanted to show that there can possibly be a minimally invasive way to add this functionality. But I thank you for raising the concerns of finding false positives in the markup - regex and the notion that `markup` has a fundamental behaviour change, that was definitely not my intention.

Reply:
That's fine - don't worry about it (Ruby not Rust though :)
I will take a look during the weekend.

Reply:
I really wrote Rust... jesus... shows how much I'm familiar with Ruby I guess.
Thanks for #127! this is fantastic. Thanks a ton.